plugins/nlp/nlp_eng.cpp

```diff
@@ -0,0 +1,75 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2021 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "nlp_eng.hpp"
+#include "nlp_engine.h"
+
+static NLP_ENGINE *nlpEngine = NULL;
+
+NLPEng::NLPEng(string workingFolderPath) {
+    if (!workingFolderPath.empty()) {
+        workingFolder = workingFolderPath;
+    } else {
+        workingFolder.clear();
+    }
+
+#ifdef NLP_DEBUG    
+    ofstream handle;
+    handle.open(NLP_DEBUG_FILE, std::ofstream::out | std::ofstream::app);
+    handle << "[workingFolder: " << workingFolder << "]" << endl;
+    handle.close();
+#endif
+}
+
+NLPEng::~NLPEng() {
+    if (nlpEngine != NULL) {
+        delete nlpEngine;
+    }
+}
+
+#include <sstream>
+
+int NLPEng::nlpEngAnalyze(const char *analyzerName, const char *inputText, ostringstream &sso)
+{
+    if (nlpEngine == NULL) {
+        nlpEngine = new NLP_ENGINE(workingFolder);
+    }
+    istrstream ssi(inputText);
+
+#ifdef NLP_DEBUG
+    clock_t s_time, e_time;
+    s_time = clock();
+#endif
+
+    nlpEngine->analyze((char *)analyzerName,&ssi,&sso);
```

Comment:
This cast shouldn't really be here - it implies the analyze() function is allowed to modify the analyzerName.
If this is the case it may well cause problems.  If it isn't the parameter should readlly be a const char *.
Since it is a separate library the cast is probably ok, as long as there is a clear comment before hand that the argument is not modified, but the prototype is wrong.